src/Kernel/BlockClosure.class.st

```diff
@@ -297,6 +297,14 @@ BlockClosure >> home [
 	^ outerContext ifNotNil: [ outerContext home ]
 ]
 
+{ #category : #accessing }
+BlockClosure >> homeMethod [
+	"return the home method. If no #home is available due to no outerContext, use the compiledBlock"
+	^ (self home
+		   ifNotNil: [ :homeContext | homeContext ]
+		   ifNil: [ self compiledBlock ]) method
```

Comment:
What is the reason to not always use "self compiledBlock method"?

Reply:
it is  slower, but I wonder if we care 

```
block := [[]]value. 

[block homeMethod ] bench "'53052503.698 per second'" 
[block homeMethod2 ] bench "'28656341.263 per second'"  

53052503.698/28656341.263 "1.8513355634307516"  
```